benchmark.cc

```diff
@@ -0,0 +1,187 @@
+#include "Product.h"
+#include "Chained.h"
+#include "Open.h"
+#include "StdMap.h"
+#include "StdUnorderedMap.h"
+
+#include <iostream>
+#include <vector>
+#include <chrono>
+#include <unistd.h>
+
+enum MAP_TYPE {STDMAP = 0, CHAINEDMAP = 1, OPENMAP = 2, STDUNORDEREDMAP = 3};
+
+void read_args(int, char**);
+
+//command line options
+MAP_TYPE mapType = STDMAP;
+uint64_t verbose = 0;
+uint64_t mapSize = 100000;
+double load = 0.7;
+
+struct productsArr{
+    std::string* products;
+    uint64_t size;
+};
+
+struct graphStat{
+    graphStat(MAP_TYPE t, double l, uint64_t n) {
+        mapType = t;
+        loadFactor = l;
+        numberOfItems = n;
+        msToFill = -1;
+        msToRead = -1;
+    }
+    MAP_TYPE mapType;
+    double loadFactor;
+    uint64_t numberOfItems;
+    uint64_t msToFill;
+    uint64_t msToRead;
```

Comment:
When benchmarking, size your tests to run for 30s or a minute.  More for larger systems.